net/libreswan/Makefile

```diff
@@ -0,0 +1,161 @@
+#
+# Copyright (C) 2019 Lucian Cristian
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+
+include $(TOPDIR)/rules.mk
+#include $(INCLUDE_DIR)/kernel.mk
+
+PKG_NAME:=libreswan
+PKG_VERSION:=3.27
+PKG_RELEASE:=1
+
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
+PKG_SOURCE_URL:=https://download.libreswan.org/
+PKG_HASH:=ead07dd701116094b483dc57e54e2a5ee9a06d3982bb142260bcbf3d1faf7b82
+
+PKG_LICENSE:=GPL-2.0
+PKG_MAINTAINER:=Lucian Cristian <lucian.cristian@gmail.com>
+
+PKG_BUILD_PARALLEL:=1
+PKG_INSTALL:=1
+
+include $(INCLUDE_DIR)/package.mk
+
+define Package/libreswan/Default
+  TITLE:=Libreswan
+  URL:=http://www.libreswan.org/
+endef
+
+define Package/libreswan/Default/description
+ Libreswan is an IPsec implementation for Linux.
+endef
+
+define Package/libreswan
+$(call Package/libreswan/Default)
+  SUBMENU:=VPN
+  SECTION:=net
+  CATEGORY:=Network
+  DEPENDS:= +kmod-libreswan +libnss +librt +libevent2 +libevent2-pthreads \
+	    +ip-full
+  PROVIDES:=openswan
+  CONFLICTS:=strongswan
+  TITLE+= (daemon)
+endef
+
+define Package/libreswan/description
+$(call Package/libreswan/Default/description)
+ This package contains the Libreswan user-land daemon.
+endef
+
+define KernelPackage/libreswan
+$(call Package/libreswan/Default)
+  SUBMENU:=Network Support
+  TITLE+= (kernel module)
+  FILES:=$(PKG_BUILD_DIR)/modobj*/ipsec.$(LINUX_KMOD_SUFFIX)
+  DEPENDS:= +kmod-crypto-authenc +kmod-crypto-hash +kmod-ipt-ipsec +iptables-mod-ipsec \
+	    +kmod-ipsec +kmod-ipsec4 +kmod-crypto-rng +IPV6:kmod-ipsec6
+endef
+
+define KernelPackage/libreswan/description
+$(call Package/libreswan/Default/description)
+ This package contains the Libreswan kernel module.
+endef
+COMMON_MAKE = \
+    $(MAKE) -C $(PKG_BUILD_DIR) \
+    $(TARGET_CONFIGURE_OPTS) \
+    LINUX_RELEASE="$(LINUX_RELEASE)" \
+    KERNELSRC="$(LINUX_DIR)" \
+    CROSS_COMPILE="$(TARGET_CROSS)" \
+    IPSECDIR="/usr/lib/ipsec" \
+    INC_USRLOCAL="/usr" \
+    INC_RCDEFAULT="/etc/init.d" \
+    MODPROBE="/sbin/insmod" \
+    FINALRUNDIR="/var/run/pluto" \
+    FINALBINDIR="/usr/libexec/ipsec" \
+    FINALLIBEXECDIR=/usr/libexec/ipsec \
+    LDFLAGS="$(TARGET_LDFLAGS)" \
+    DESTDIR="$(PKG_INSTALL_DIR)" \
+    WERROR_CFLAGS=" " \
+    USE_LINUX_AUDIT=false \
+    USE_LABELED_IPSEC=false \
+    USE_DNSSEC=false \
+    USE_NM=false \
+    USE_LIBCURL=false \
+    USE_GLIBC_KERN_FLIP_HEADERS=true \
+    USE_XAUTHPAM=false \
+    USE_FIPSCHECK=false \
+    USE_LIBCAP_NG=false \
+    USE_SYSTEMD_WATCHDOG=false
+
+TARGET_CFLAGS+= -Wno-error=format-nonliteral
```

Comment:
I hate this so much. Is it possible to backport a patch that fixes the error or are there too many errors to patch?

Reply:
this is the only major error that I saw
they recommend using `WERROR_CFLAGS=" "` on cross and in some point of the compile this is not taken into account so I added them one more

Reply:
this is some part of the output

```
home/build/proxy/staging_dir/toolchain-x86_64_gcc-7.4.0_musl/include/fortify/stdio.h: In function 'snprintf':
/home/build/proxy/staging_dir/toolchain-x86_64_gcc-7.4.0_musl/include/fortify/stdio.h:101:2: warning: format not a string literal, argument types not checked [-Wformat-nonliteral]
  return __orig_snprintf(__s, __n, __f, __builtin_va_arg_pack());
  ^~~~~~
/home/build/proxy/staging_dir/toolchain-x86_64_gcc-7.4.0_musl/include/fortify/stdio.h: In function 'sprintf':
/home/build/proxy/staging_dir/toolchain-x86_64_gcc-7.4.0_musl/include/fortify/stdio.h:110:3: warning: format not a string literal, argument types not checked [-Wformat-nonliteral]
   __r = __orig_snprintf(__s, __b, __f, __builtin_va_arg_pack());
   ^~~
/home/build/proxy/staging_dir/toolchain-x86_64_gcc-7.4.0_musl/include/fortify/stdio.h:114:3: warning: format not a string literal, argument types not checked [-Wformat-nonliteral]
   __r = __orig_sprintf(__s, __f, __builtin_va_arg_pack());
   ^~~

```

Reply:
Oh that error. Hmm no good solution to that one. Marking as resolved.